client/src/component/Header.js

```diff
@@ -66,13 +66,14 @@ export default function Header({ links, isAdminPage }) {
     (async () => {
       setAuthUrl(
         await (isAuthenticated
-          ? buildLogoutUrl()
+          ? buildLogoutUrl() // this not to be waited
           : buildAuthorizeUrl({
+              // this returns a promise
               audience: process.env.REACT_APP_AUDIENCE,
             }))
       );
     })();
-  }, [isAuthenticated]);
+  }, [isAuthenticated]); // when isAuthenticated changes logIn changes to logOut viseversa
```

Comment:
I'm still pretty sure you don't need this effect 😄 

Also, looking at the docs, it looks like you should be using `loginWithRedirect`.

https://auth0.com/docs/quickstart/spa/react/01-login#add-login-to-your-application